File: Python/jan29/gridland_metro.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'gridlandMetro' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER n
#  2. INTEGER m
#  3. INTEGER k
#  4. 2D_INTEGER_ARRAY track
#

def gridlandMetro(n, m, k, track):
    
    d = {}             
    total = n*m
    
    for r, c1, c2 in track:
        if r not in d:
            d[r] = [c1, c2]
            
        elif c1>d[r][1]:
            total -= c2-c1+1
        
        elif c2>d[r][1]:
            d[r][1] = c2
    
    track = 0
    for r in d.keys():
        track += d[r][1]-d[r][0]+1
    
    return total-track

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    first_multiple_input = input().rstrip().split()

    n = int(first_multiple_input[0])

    m = int(first_multiple_input[1])

    k = int(first_multiple_input[2])

    track = []

    for _ in range(k):
        track.append(list(map(int, input().rstrip().split())))

    result = gridlandMetro(n, m, k, track)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Java/jan29/gridland_metro.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'gridlandMetro' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts following parameters:
     *  1. INTEGER n
     *  2. INTEGER m
     *  3. INTEGER k
     *  4. 2D_INTEGER_ARRAY track
     */

    public static long gridlandMetro(int n, int m, int k, List<List<Integer>> track) 
    {
        HashMap<Integer, int[]> d = new HashMap<>();
        long total = (long)n*m;

        for(int i=0; i<track.size(); i++)
        {
            int r = track.get(i).get(0);
            int c1 = track.get(i).get(1);
            int c2 = track.get(i).get(2);
            
            if(!d.containsKey(r))
            {
                int temp[] = {c1,c2};
                d.put(r, temp);
            }
            else if(c1 > d.get(r)[1])
                total -= (c2-c1+1);
                
            else if (c2 > d.get(r)[1])
                d.get(r)[1] = c2;
        }

        long tracks = 0;
        for(int r : d.keySet())
            tracks += d.get(r)[1] - d.get(r)[0] + 1;
        
        return(total - tracks);
        
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String[] firstMultipleInput = bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");

        int n = Integer.parseInt(firstMultipleInput[0]);

        int m = Integer.parseInt(firstMultipleInput[1]);

        int k = Integer.parseInt(firstMultipleInput[2]);

        List<List<Integer>> track = new ArrayList<>();

        IntStream.range(0, k).forEach(i -> {
            try {
                track.add(
                    Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                        .map(Integer::parseInt)
                        .collect(toList())
                );
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        });

        long result = Result.gridlandMetro(n, m, k, track);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
